protocol/x/vault/keeper/msg_server_allocate_to_vault.go

```diff
@@ -0,0 +1,42 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	assetstypes "github.com/dydxprotocol/v4-chain/protocol/x/assets/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// AllocateToVault allocates funds from main vault to a vault.
+func (k msgServer) AllocateToVault(
```

Comment:
What happens if the vault doesn't exist in state?

Reply:
i was thinking that the operator needs to separately set vault params to have a vault exist in state

a smoother experience is probably to set vault params with `stand_by` status here during allocation if the allocated-to vault doesn't exist

Reply:
>i was thinking that the operator needs to separately set vault params to have a vault exist in state

Is this validated anywhere? What happens if the operator doesn't make the vault exist?

Reply:
not validated so I'd vote to set vault params with `stand_by` status during allocation. wdyt?